### PR TITLE
Environment creation

### PR DESCRIPTION
**Pull request by enformatique11**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,32 @@ dev:
...
+	$(eval ENV_TAG=pre-prod)
+
+.PHONY: production
+production:
+	$(eval DEPLOY_ENV=production)
+	$(eval AZURE_SUBSCRIPTION=s165-teachingqualificationsservice-production)
+	$(eval RESOURCE_NAME_PREFIX=s165p01)
+	$(eval ENV_SHORT=pd)
+	$(eval ENV_TAG=prod)
+	$(eval AZURE_BACKUP_STORAGE_ACCOUNT_NAME=s165p01rsmdbbackuppd)
```
Comment (NickGraham101) [inline Makefile:40]: these 2 properties are using `rsm` rather than `trngen`

File: terraform/app.tf [hunk 2]
```diff
@@ -27,10 +29,12 @@ resource "azurerm_linux_web_app" "web_app" {
   dynamic "sticky_settings" {
     for_each = var.enable_blue_green ? [1] : []
     content {
-      app_setting_names = keys(data.azurerm_linux_web_app.web_app.app_settings)
+      app_setting_names = keys(data.azurerm_linux_web_app.web_app[0].app_settings)
     }
   }
-  app_settings = var.enable_blue_green ? data.azurerm_linux_web_app.web_app.app_settings : local.trngen_env_vars
+  # app_settings = local.trngen_env_vars
```
Comment (NickGraham101) [inline terraform/app.tf:35]: Remove commented out line

File: terraform/variables.tf [hunk 3]
```diff
@@ -32,7 +32,7 @@ variable "resource_group_name" {
 
 variable "postgres_flexible_server_sku" {
   type    = string
-  default = "B_Standard_B1ms"
+  default = "GP_Standard_D2s_v3"
```
Comment (NickGraham101) [inline terraform/variables.tf:35]: I don't think we want to change the default, the cheaper SKU should be sufficient for dev & test

File: terraform/workspace_variables/preprod.tfvars.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "environment_name": "preprod",
+  "key_vault_name": "s165t01-trngen-pp-kv",
+  "resource_group_name": "s165t01-trngen-pp-rg",
+  "storage_account_name": "s165t01trngentfstatepp",
+  "resource_prefix": "s165t01",
+  "enable_blue_green": false,
```
Comment (NickGraham101) [inline terraform/workspace_variables/preprod.tfvars.json:7]: Should be true

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -87,21 +86,43 @@ jobs:
...
+        with:
+          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+          environment_name: dev
+          image_tag: ${{ needs.build.outputs.image_tag }}
+
   deploy:
     name: Deploy to ${{ matrix.environment }} environment
     runs-on: ubuntu-latest
-    if: github.event_name == 'pull_request' && github.event.pull_request.draft == false
+    if: github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch'
```
[the remark this answers is not in the document]
Comment (enformatique11) [reply .github/workflows/build.yml:114]: fixed

File: terraform/output.tf [hunk 6]
```diff
@@ -16,4 +16,8 @@ output "blue_green" {
 
 output "service_sku_size" {
   value = var.app_service_plan_sku_size
-  }
+}
+
+output "high_availability" {
```
[the remark this answers is not in the document]
Comment (enformatique11) [reply terraform/output.tf:21]: no- removed

File: terraform/variables.tf (hunk 3 above)
Comment (enformatique11) [reply terraform/variables.tf:35]: changed it back to B_Standard_B1ms

File: terraform/workspace_variables/preprod.tfvars.json (hunk 4 above)
Comment (enformatique11) [reply terraform/workspace_variables/preprod.tfvars.json:7]: changed to true

File: terraform/workspace_variables/production.tfvars.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "environment_name": "production",
+  "key_vault_name": "s165p01-trngen-pd-kv",
+  "resource_group_name": "s165p01-trngen-pd-rg",
+  "storage_account_name": "s165p01trngentfstatepd",
+  "resource_prefix": "s165p01",
+  "enable_blue_green": false,
```
[the remark this answers is not in the document]
Comment (enformatique11) [reply terraform/workspace_variables/production.tfvars.json:7]: changed to true

File: Makefile (hunk 1 above)
Comment (enformatique11) [reply Makefile:40]: fixed

**Comment by enformatique11** [comment]
[Postgres HA](https://trello.com/c/iqCngtvN/373-postgres-ha)

**Comment by enformatique11** [comment]
[App Service Instances](https://trello.com/c/NkksELa3/372-app-service-instances)

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**